cmd/bosun/main.go

```diff
@@ -137,19 +143,69 @@ func main() {
 			}))
 			slog.Infoln("readonly relay at", ts.URL, "to", tsdbHost)
 			tsdbHost, _ = url.Parse(ts.URL)
-			c.TSDBHost = tsdbHost.Host
+			sysProvider.SetTSDBHost(tsdbHost.Host)
 		}
 	}
-	if c.InternetProxy != "" {
-		web.InternetProxy, err = url.Parse(c.InternetProxy)
+	if sysProvider.GetInternetProxy() != "" {
+		web.InternetProxy, err = url.Parse(sysProvider.GetInternetProxy())
 		if err != nil {
 			slog.Fatalf("InternetProxy error: %s", err)
 		}
 	}
-	if *flagQuiet {
-		c.Quiet = true
+	cmdHook, err := conf.MakeSaveCommandHook(sysProvider.GetCommandHookPath())
+	if err != nil {
+		slog.Fatal(err)
+	}
+	ruleProvider.SetSaveHook(cmdHook)
```

Comment:
The file is saved, and then the hook is called. If the hook fails, the orginal file is restored:

```
if c.saveHook != nil {
		err := c.callSaveHook(c.Name, user, message, args...)
		if err != nil {
			sErr := c.SaveConf(c)
			restore := "successful"
			if sErr != nil {
				restore = sErr.Error()
			}
			return fmt.Errorf("failed to call save hook: %v. Restoring config: %v", err, restore)
		}
	}
	err = c.reload()
```

Also the hook is passed arguments that reload doesn't need to be aware of. But maybe there is a better way?